analyzers/tests/SonarAnalyzer.UnitTest/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs

```diff
@@ -289,52 +296,64 @@ public void BranchedPropagationChain_LiveIn_LiveOut()
+            var context = new Context(code);
+            context.Validate(context.Cfg.EntryBlock, new LiveIn(/*FIXME: Will be removed*/ "variable", "boolParameter"), new LiveOut(/*FIXME: Will be removed*/ "variable", "boolParameter"));
+            context.Validate(context.Block("Method(nameof(variable));"));
         }
 
         [TestMethod]
-        public void ProcessIdentifier_LocalScopeSymbol_LiveIn()
+        public void ProcessLocalReference_LocalScopeSymbol_LiveIn()
         {
             var code = @"
 var variable = 42;
 if (boolParameter)
     return;
 Method(intParameter, variable);";
             var context = new Context(code);
-            throw new System.NotImplementedException();
-            //context.Validate(context.Block<SimpleBlock>(), new LiveIn("intParameter", "variable"));
+            context.Validate(context.Cfg.EntryBlock, new LiveIn( /*FIXME: Will be removed*/ "variable", "boolParameter", "intParameter"), new LiveOut(/*FIXME: Will be removed*/ "variable", "boolParameter", "intParameter"));
+            context.Validate(context.Block("Method(intParameter, variable);"), new LiveIn("intParameter", "variable"));
         }
 
         [TestMethod]
-        public void ProcessIdentifier_GlobalScopeSymbol_NotLiveIn_NotLiveOut()
+        public void ProcessLocalReference_GlobalScopeSymbol_NotLiveIn_NotLiveOut()
         {
             var code = @"
 var s = new Sample();
 Method(field, s.Property);";
+            throw new NotImplementedException(); // Depends on variable declarator processing
             new Context(code).ValidateAllEmpty();
         }
 
         [TestMethod]
-        public void ProcessIdentifier_UndefinedSymbol_NotLiveIn_NotLiveOut()
+        public void ProcessLocalReference_UndefinedSymbol_NotLiveIn_NotLiveOut()
         {
             var code = @"Method(undefined);";
             new Context(code).ValidateAllEmpty();
         }
 
         [TestMethod]
-        public void ProcessIdentifier_RefOutArgument_NotLiveIn_LiveOut()
+        public void ProcessLocalReference_UsedAsOutArgument_NotLiveIn()
         {
             var code = @"
-outParameter = true;
-Method(outParameter, refParameter);";
-            new Context(code).ValidateAllEmpty();
+var refVariable = 42;
+var outVariable = 42;
+if (boolParameter)
+    return;
+Main(true, 0, out outVariable, ref refVariable);";
+            var context = new Context(code);
+            context.Validate(context.Cfg.EntryBlock, new LiveIn(/*FIXME: Will be removed*/"refVariable", /*FIXME: Will be removed*/"outVariable", "boolParameter"), new LiveOut(/*FIXME: Will be removed*/"refVariable", /*FIXME: Will be removed*/"outVariable", "boolParameter"));
+            context.Validate(context.Block("Main(true, 0, out outVariable, ref refVariable);"), new LiveIn("refVariable"));
         }
 
         [TestMethod]
-        public void ProcessIdentifier_Assigned_NotLiveIn_LiveOut()
+        public void ProcessLocalReference_Assigned_NotLiveIn_LiveOut()
```

Comment:
`ProcessIdentifier_*` refers to Sonar-specific method, while `ProcessLocalReference_*` refers to Roslyn-specific implementation method. Almost, because I renamed them to `ProcessParameterOrLocalReference` on Roslyn side :/